Week1/homework/app.js

```diff
@@ -1,11 +1,303 @@
+
+      // if gender is female then return she
+      const genderFinder = person => (person === 'female' ? 'she' : 'he');
+
+      author.innerHTML = `by ${authorName}`;
+      li.appendChild(author);
+      const authorInfo = document.createElement('p');
+      authorInfo.innerHTML = `${authorName} is originally from ${authorCountry}, ${genderFinder(
+        authorGender,
+      )} has died at the age of ${authorAge}.`;
+      li.appendChild(authorInfo);
+
+      // Add Book info
+      const bookLanguage = myBooks[key].book_info.language.mainLanguage;
+      const publishYear = myBooks[key].book_info.publication;
+      const translateNum = myBooks[key].book_info.language.translations;
+
+      const generalInfo = document.createElement('p');
+      generalInfo.innerHTML = `The book was originally published in ${bookLanguage}, in ${publishYear}. Later it has been translated to ${translateNum}. `;
+      li.appendChild(generalInfo);
+
+      // Add plot
+      const plot = document.createElement('h3');
+      plot.innerHTML = 'Main Plot: ';
+      li.appendChild(plot);
+      const bookPlot = document.createElement('p');
+      bookPlot.innerHTML = `'${myBooks[key].plot}'`;
+      li.appendChild(bookPlot);
+    }
+  }
+};
+
+document.getElementById('bookInfoList').appendChild(ul1);
+
+// 1.7
+
+const bookCovers = {
+  'War and Peace': './images/war_and_peace.jpeg',
+  'The Great Gatsby': './images/great_gatsby.jpg',
+  'Great Expectations': './images/great_expectations.jpg',
+  'Crime and Punishment': './images/crime_and_punishment.jpeg',
+  'Moby Dick': './images/moby_dick.jpg',
+  Hamlet: './images/hamlet.jpg',
+  'Madame Bovary': './images/madame_bovary.jpg',
+  'Jane Eyre': './images/jane_eyre.jpg',
+  'Pride and Prejudice': './images/pride_and_prejudice.jpg',
+  'Anna Karenina': './images/anna_karenina.jpeg',
+};
+
+// 1.8 I could not do this with Object.keys(objectName) I hope I can learn it from someone
+
+const ul2 = document.createElement('ul');
+
+const getBookCovers = () => {
+  for (const key in bookCovers) {
+    if (Object.prototype.hasOwnProperty.call(myBooks, key)) {
+      // Add title
+      const title = document.createElement('li');
+      title.innerHTML = key;
+      ul2.appendChild(title);
+      document.body.appendChild(ul2);
+      // Add image
+      const image = document.createElement('img');
+      image.src = bookCovers[key];
+      image.alt = key;
+      ul2.appendChild(image);
+      // Add button
+      const btn = document.createElement('button');
+      btn.innerHTML = 'Add to cart';
+      ul2.appendChild(btn);
+    }
+  }
+  document.getElementById('coverList').appendChild(ul2);
+};
+
+window.addEventListener('load', getBookInfo);
+window.addEventListener('load', getBookCovers);
 
-  // Replace with your own code
-  console.log(bookTitles);
-}
+// I used element.attribute instead of setAttribute() throughout this homework
+// it feels easier to do it that way, but is it bad practice?
```

Comment:
Using `element.attribute` is perfectly fine in practice!

Reply:
Okay that's good to know thank you for the information